fido/src/main/java/com/yubico/yubikit/fido/ctap/Hkdf.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.fido.ctap;
+
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import javax.annotation.Nullable;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implements HMAC-based Extract-and-Expand Key Derivation Function (HKDF)
+ *
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc5869">rfc5869</a>
+ */
+public class Hkdf {
```

Comment:
I would prefer this class to be package-private to avoid making it part of the public API.